src/control/server/superblock.go

```diff
@@ -154,7 +154,7 @@ func (srv *IOServerInstance) CreateSuperblock(msInfo *mgmtInfo) error {
 	}
 	srv.setSuperblock(superblock)
 	srv.log.Debugf("creating %s: (rank: %d, uuid: %s)",
-		srv.superblockPath(), superblock.Rank, superblock.UUID)
+		srv.superblockPath(), *superblock.Rank, superblock.UUID)
```

Comment:
Hmm. I think that this is potentially unsafe. As the code is currently written, we can't have a nil rank in the superblock, but my concern is that things may change over time and this could eventually result in a panic if somehow a nil rank gets into the superblock struct.

The safer approach, I think, would be to adjust the Rank.String() method to have a pointer receiver and check there for a nil pointer. We could also check for the special NilRank value and print a string to distinguish between a nil pointer and NilRank.

What do you think?

Reply:
when printing the dereferenced Rank which has a stringer implementation already I don't think we would ever get a panic, have added a check for nil Rank

Reply:
Right, what I meant is that the code as it is right now is safe, because there is a check above it to make sure that the rank isn't nil. But if things change and that check is moved elsewhere or removed for whatever reason, then dereferencing a nil pointer would result in a panic. If we move to using the stringer no matter what, and we change the stringer to having a pointer receiver, then we can always safely include the rank in a formatted statement.